builds/jazz_azure-create-service/components/ResourceFactory.js

```diff
@@ -220,27 +219,55 @@ module.exports = class ResourceFactory {
   }
 
 
-  async createOrUpdateApiGatewayWithSwaggerJson(serviceName, apiId, swagger, basePath = "api", resourceGroupName = this.resourceGroupName) {
+  async createOrUpdateApiGatewayWithSwaggerJson(serviceName, apiId, swagger, appName, basePath = "api", resourceGroupName = this.resourceGroupName) {
     const fs = require("fs");
     let text = fs.readFileSync(swagger);
     var obj = JSON.parse(text);
+    obj = this.updateSwagger(obj, appName)
 
     let parameters = {
-      "contentFormat": "swagger-json",
-      "contentValue": JSON.stringify(obj),
-      "path": basePath
+        "format": "swagger-json",
+        "value": JSON.stringify(obj),
+        "path": basePath
     }
+
+    let dataParams = {
+      "description": "Sample Description API",
+      "displayName": appName,
+      "serviceUrl": "http://dev-cloud-api.corporate.t-mobile.com/api",
+      "path": appName,
+      "protocols": [
+        "https"
+      ]
+    }
+
     let client = await this.factory.getResource("ApiManagementClient");
+    let dataValue = await client.api.createOrUpdate(resourceGroupName, serviceName, apiId, dataParams, null);
     let result = await client.api.createOrUpdate(resourceGroupName, serviceName, apiId, parameters, null);
     this.serviceName = serviceName;
     return this.withStack(result);
   }
 
+  updateSwagger(swaggerFile, appName){
```

Comment:
Can you add comments there to indicate what values in swagger are removed or replaced so it is easier to follow?

Reply:
added description of the changes to swagger